File: javav2/example_code/forecast/src/main/java/com/example/forecast/DeleteDataset.java
//snippet-sourcedescription:[DeleteDataset.java demonstrates how to delete a data set that belongs to the Amazon Forecast service.]
//snippet-keyword:[AWS SDK for Java v2]
//snippet-service:[Amazon Forecast]

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.example.forecast;

// snippet-start:[forecast.java2.delete_forecast_dataset.import]
import software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider;
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.forecast.ForecastClient;
import software.amazon.awssdk.services.forecast.model.DeleteDatasetRequest;
import software.amazon.awssdk.services.forecast.model.ForecastException;
// snippet-end:[forecast.java2.delete_forecast_dataset.import]


/**
 * Before running this Java V2 code example, set up your development environment, including your credentials.
 *
 * For more information, see the following documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */
public class DeleteDataset {

    public static void main(String[] args) {

        final String usage = "\n" +
            "Usage:\n" +
            "    <datasetARN> \n\n" +
            "Where:\n" +
            "    datasetARN - The ARN of the data set to delete. \n\n" ;

        if (args.length != 1) {
            System.out.println(usage);
            System.exit(1);
        }

        String datasetARN = args[0];
        Region region = Region.US_WEST_2;
        ForecastClient forecast = ForecastClient.builder()
            .region(region)
            .credentialsProvider(ProfileCredentialsProvider.create())
            .build();

        deleteForecastDataSet(forecast, datasetARN);
        forecast.close();
    }

    // snippet-start:[forecast.java2.delete_forecast_dataset.main]
    public static void deleteForecastDataSet(ForecastClient forecast, String myDataSetARN) {

        try {
            DeleteDatasetRequest deleteRequest = DeleteDatasetRequest.builder()
                .datasetArn(myDataSetARN)
                .build();

            forecast.deleteDataset(deleteRequest);
            System.out.println("The Data Set was deleted") ;

        } catch (ForecastException e) {
            System.err.println(e.awsErrorDetails().errorMessage());
            System.exit(1);
        }
    }
    // snippet-end:[forecast.java2.delete_forecast_dataset.main]
}

File: kotlin/services/forecast/src/main/kotlin/com/kotlin/forecast/DeleteDataset.kt
// snippet-sourcedescription:[DeleteDataset.kt demonstrates how to delete a data set that belongs to the Amazon Forecast service.]
// snippet-keyword:[AWS SDK for Kotlin]
// snippet-service:[Amazon Forecast]

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.forecast

// snippet-start:[forecast.kotlin.delete_forecast_dataset.import]
import aws.sdk.kotlin.services.forecast.ForecastClient
import aws.sdk.kotlin.services.forecast.model.DeleteDatasetRequest
import kotlin.system.exitProcess
// snippet-end:[forecast.kotlin.delete_forecast_dataset.import]

/**
Before running this Kotlin code example, set up your development environment,
including your credentials.

For more information, see the following documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */

suspend fun main(args: Array<String>) {

    val usage = """
    Usage:
        <dataSetARN>  

    Where:
        dataSetARN - The ARN of the data set to delete. 
       """

    if (args.size != 1) {
        println(usage)
        exitProcess(0)
    }

    val dataSetARN = args[0]
    deleteForecastDataSet(dataSetARN)
}

// snippet-start:[forecast.kotlin.delete_forecast_dataset.main]
suspend fun deleteForecastDataSet(myDataSetARN: String?) {

    val request = DeleteDatasetRequest {
        datasetArn = myDataSetARN
    }

    ForecastClient { region = "us-west-2" }.use { forecast ->
        forecast.deleteDataset(request)
        println("$myDataSetARN data set was deleted")
    }
}
// snippet-end:[forecast.kotlin.delete_forecast_dataset.main]
